### PR TITLE
Fix tox support and only test supported versions of Python

### LINKED ISSUE #1: Python 3 support for PyPDF4
Sorry if I had no other means of putting the question. A few numbered questions about the future of PyPDF4 and Python releases:

1. Should the new code be developed in Python 3?
2. Shall the contributors try to make it compatible with Python 2 or concern themselves with Python 3 only?
3. Does the old code need to retain its compliance with Python 2 only, Python 3 only, or both Python 2 and Python 3?

If the issues page is deemed not very apt for this kind of discussion, I accept to move it elsewhere.

### LINKED ISSUE #36: tox cannot run the unit tests
tox is no longer able to run the unit tests. This is because `tox.ini` has not been updated.

In addition, unsupported versions of Python are currently targeted for testing. PyPDF4 needs to drop support for Python 2.6, 3.3, and 3.4, and needs to add support for 3.6 and 3.7.

Finally, a coverage report needs to be generated so that developers can see what code is being run by the unit tests.

### PR DESCRIPTION
**Pull request by kurtmckee**
This patch:

* Fixes tox support so that all versions of Python can be tested in one command
* Drops support for Python 2.6, 3.3, and 3.4.
* Adds support for Python 3.6 and 3.7.
* Adds HTML coverage reporting so that it's easy to see what is and is not getting tested.

Note that the unit tests are currently failing on **all** Python versions. I will be working to resolve this in a separate merge request.

Closes #1
Fixes #36

### CODE REVIEW

**Comment by claird** [comment]
This sounds wonderful, Kurt.  It's going to be at least a couple of days
before I have time to examine it.  I apologize for the delay.




Cameron Laird, vice president
We make computers work for people.

**Comment by kurtmckee** [comment]
Thanks, Cameron! I completely understand having limited time to review things. I hope your week is going well! =)

**Comment by newnone** [comment]
@claird you can leave the review up to me. If it's ok, I can merge right away.

**Review by acsor — APPROVED** [review]

**Comment by newnone** [comment]
I'd like to have a few days to go before the changes I have approved are integrated. I do not think such a slow pace is beneficial for the overall project.

**Comment by newnone** [comment]
I merged manually this PR, minus my modification from [c4fbd73d9bf51d1629f1230414ad49aa711cdc03](https://github.com/claird/PyPDF4/pull/37/commits/c4fbd73d9bf51d1629f1230414ad49aa711cdc03). I wonder whether I can have GitHub recognize the merge automatically or close myself this ticket.

**Pull request closed without merging**
